File: tests/receiver.py
# simple testing script to connect a socket to the server
# receive data and internal errors

import json
from socketIO_client import SocketIO


def on_data(data):
    print json.dumps(data)


def on_error(err):
    print json.dumps(err)


def open_client():
    socketIO = SocketIO("streaming.plenar.io")

    socketIO.on('data', on_data)
    socketIO.on('internal_error', on_error)
    socketIO.wait()

File: tests/receiver.js
/**
 * simple testing script to connect a socket to the server
 * prints data, transport upgrades, and internal errors
 */

var io = require('socket.io-client');

var socket = io.connect('http://streaming.plenar.io?network=plenario_development');

socket.on('data', function (data) {
    console.log(data);
});
socket.io.engine.on('upgrade', function(transport){
    console.log('upgraded to:');
    console.log(transport.query.transport);
});
socket.on('internal_error', function (err) {
    console.log(err);
});